crates/shared/src/order_validation.rs

```diff
@@ -214,6 +214,26 @@ pub trait LimitOrderCounting: Send + Sync {
     async fn count(&self, owner: Address) -> Result<u64>;
 }
 
+#[derive(Clone, Debug, clap::ValueEnum)]
+pub enum SameTokensPolicy {
+    Disallow,
+    AllowSell,
+    // Allow, TODO: Allow sell and buy orders with the same tokens (https://github.com/cowprotocol/services/issues/3963)
+}
+
+impl SameTokensPolicy {
+    fn validate_same_sell_and_buy_token(&self, order: &PreOrderData) -> bool {
+        if order.sell_token != order.buy_token {
+            return true;
+        }
+        matches!(
+            (self, order.kind),
+            /* (Self::Allow, _) | To be implemented in https://github.com/cowprotocol/services/issues/3963 */
```

Comment:
nano-nit: why not a regular comment?
```suggestion
            // (Self::Allow, _) | To be implemented in https://github.com/cowprotocol/services/issues/3963
```